ttnn/cpp/ttnn/operations/eltwise/binary/device/broadcast_height_and_width_multi_core_program_factory.cpp

```diff
@@ -41,15 +40,13 @@ BinaryDeviceOperation::BroadcastHeightAndWidthMultiCore::create(
     auto& output = tensor_return_value;
     auto bcast_math = binary_op_type_to_bcast_op_math(operation_attributes.binary_op_type);
     const auto ashape = a.get_legacy_shape();
-    const auto bshape = b.get_legacy_shape();
+    const auto bshape = b.has_value() ? b->get_legacy_shape() : Shape{1, 1};
```

Comment:
Can we please instead rely on the volume? 
If the volume is 1 - it is a scalar. 

New Tensor soon allows tensors of rank 0 or 1. 
Below code seem to rely that bshape[-1] and bshape[-2] is accessible - that will have to be checked now, but after the upcoming change it is possible to access those values - it will return 1.

Also, please use `get_padded_shape` instead of get_legacy_shape.